js/style/style.js

```diff
@@ -644,7 +644,7 @@ Style.prototype = util.inherit(Evented, {
         }
 
         var includedSources = {};
-        if (params.layers) {
+        if (params && params.layers) {
```

Comment:
@lucaswoj In the test [here](https://github.com/mapbox/mapbox-gl-js/pull/2979/files#diff-277bb5e9f409aee234cc524db362a55bR567) I hit the error `Error: TypeError: Cannot read property 'layers' of undefined`, which led to a check to ensure `params` exists in the first place.